ui/page/wallets/wallet_settings_page.go

```diff
@@ -246,15 +261,13 @@ func (pg *WalletSettingsPage) Handle() {
 								return
 							}
 
-							pg.RefreshWindow()
 							if pg.WL.MultiWallet.LoadedWalletsCount() > 0 {
 								pg.Toast.Notify("Wallet removed")
 								pg.PopFragment()
 							} else {
-								pg.WL.Wallet.ClearListeners()
-								pg.Load.Receiver.AllWalletsDeleted <- struct{}{}
+								pg.Load.ReloadApp()
```

Comment:
`win.Loop` is now simplified to only handle actual window events, not auxiliary events like this. The method approach should achieve the same effect.